### PR TITLE
Replace all calls to strncpy with strlcpy, use strdup more, expose strlcat

### PR DESCRIPTION
**Pull request by CookiePLMonster**
# What's new

This PR improves the consistency of string copy calls, removing all uses of `strncpy` and `strcat`, at least from the code that isn't in submodules. `strlcpy` is preferable over `strncpy`, because it doesn't needlessly zero the remaining destination buffer, and it ensures null-termination.

Because `strlcat` is already used in mjs, I exposed it in the API. Please check if this is fine.

To make buffer sizes consistent, I modified the following constants to include the space for the null terminator. Please check if this is fine:
* `IBUTTON_KEY_NAME_SIZE`
* `INFRARED_MAX_BUTTON_NAME_LENGTH`
* `INFRARED_MAX_REMOTE_NAME_LENGTH`
* `MAX_TEXT_INPUT_LEN`

# Verification 

* Run a full suite of tests.

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: applications/debug/rpc_debug_app/rpc_debug_app.c [hunk 1]
```diff
@@ -60,7 +60,10 @@ static void rpc_debug_app_rpc_command_callback(const RpcAppSystemEvent* event, v
         furi_assert(event->data.type == RpcAppSystemEventDataTypeBytes);
 
         rpc_debug_app_format_hex(
-            event->data.bytes.ptr, event->data.bytes.size, app->text_store, TEXT_STORE_SIZE);
+            event->data.bytes.ptr,
+            event->data.bytes.size,
+            app->text_store,
+            sizeof(app->text_store));
```
Comment (skotopes) [inline applications/debug/rpc_debug_app/rpc_debug_app.c]: There is one thing that I don't like about sizeof: compiler will not inform you if you change char[] with char*.

**Review by skotopes — CHANGES_REQUESTED** [review]

**Comment by skotopes** [comment]
Un-draft when ready

File: applications/debug/rpc_debug_app/rpc_debug_app.c (hunk 1 above)
Comment (CookiePLMonster) [reply applications/debug/rpc_debug_app/rpc_debug_app.c]: Fair point. Should I change all the occurrences back? Or does `COUNT_OF` account for it? AFAIK it doesn't.

Comment (skotopes) [reply applications/debug/rpc_debug_app/rpc_debug_app.c]: yep, roll em back please

**Review by skotopes — APPROVED** [review]

**Pull request merged**